apps/cron/src/tasks/processGemsPayout/createSablierAirdropContract.ts

```diff
@@ -0,0 +1,151 @@
+};
+
+type Recipient = {
+  address: `0x${string}`;
+  amount: number;
+};
+
+function createCsvContent(recipients: Recipient[]) {
+  let csvContent = 'address,amount\n';
+  recipients.forEach(({ address, amount }) => {
+    csvContent += `${address.toLowerCase()},${amount}\n`;
+  });
+
+  return csvContent;
+}
+
+export async function createSablierAirdropContract({
+  adminPrivateKey,
+  tokenAddress,
+  chainId,
+  recipients,
+  campaignName,
+  tokenDecimals
+}: {
+  chainId: number;
+  adminPrivateKey: `0x${string}`;
+  tokenAddress: `0x${string}`;
+  tokenDecimals: number;
+  recipients: Recipient[];
+  campaignName: string;
+}) {
+  const publicClient = getPublicClient(chainId);
+
+  const account = privateKeyToAccount(adminPrivateKey, {
+    nonceManager
+  });
+
+  const walletClient = getWalletClient({
+    chainId,
+    privateKey: adminPrivateKey
+  });
+
+  const csvContent = createCsvContent(recipients);
+
+  const file = new File([csvContent], 'airdrop.csv', {
+    type: 'text/csv'
+  });
+
+  const formData = new FormData();
+  formData.append('data', file);
+
+  const merkleResponse = await fetch(`https://sablier-merkle-api.vercel.app/api/create?decimals=${tokenDecimals}`, {
+    method: 'POST',
+    body: formData
+  });
+
+  if (!merkleResponse.ok) {
+    const errorText = await merkleResponse.text();
+    throw new Error(`HTTP error! status: ${merkleResponse.status}, details: ${errorText}`);
+  }
+
+  const merkleData = (await merkleResponse.json()) as { root: `0x${string}`; cid: string };
+  const { root, cid } = merkleData;
+
+  const baseParams = {
+    token: tokenAddress,
+    expiration: BigInt(Math.floor(Date.now() / 1000) + 30 * 24 * 60 * 60), // 30 days
+    initialAdmin: account.address,
+    ipfsCID: cid,
+    merkleRoot: root as `0x${string}`,
+    campaignName,
+    shape: ''
+  };
+
+  const aggregateAmount = parseUnits(recipients.reduce((acc, { amount }) => acc + amount, 0).toString(), tokenDecimals);
+  const recipientCount = recipients.length;
+
+  const sablierAirdropFactoryAddress = SablierAirdropFactoryContractRecords[chainId];
+
+  if (!sablierAirdropFactoryAddress) {
+    throw new Error(`Sablier airdrop factory address not found for chainId ${chainId}`);
+  }
+
+  const { request } = await publicClient.simulateContract({
+    address: sablierAirdropFactoryAddress,
+    abi: sablierAirdropFactoryAbi,
+    functionName: 'createMerkleInstant',
+    args: [baseParams, aggregateAmount, recipientCount],
+    account
+  });
+
+  const hash = await walletClient.writeContract(request);
+
+  const receipt = await publicClient.waitForTransactionReceipt({ hash });
+
+  const createdCampaignAddress = receipt.logs[0].address as `0x${string}`;
+
+  const { request: approveCampaignRequest } = await publicClient.simulateContract({
+    address: tokenAddress,
+    abi: erc20Abi,
+    functionName: 'approve',
+    args: [createdCampaignAddress, aggregateAmount],
+    account
+  });
+
```

Comment:
As discussed, the merkle doesn't pull funds it's manually prefunded.

So this needs to be a transfer call


Please also confirm mechanics for refunding after claims expire